### PR TITLE
Initial draft of HTTP streaming documentation

### PR DESCRIPTION
**Pull request by luciferous**
Problem

It's too difficult to learn HTTP streaming by reading the source, and
the test cases in EndToEndTests.scala are too alien looking to model
real world problems on.

Solution

Documentation and examples of usage.

Result

More people know about HTTP streaming and will be comfortable to try it
out.

### CODE REVIEW

File: doc/src/sphinx/HttpStreaming.rst [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+---------------------
+
+``Request`` and ``Response`` have a convenience method for writing streams.
+
+.. code-block:: scala
+
+  def apply(req: Request): Future[Response] = {
+    val response = Response()
+    response.setChunked(true)
+    response.write("hello") before response.write("world")
```
Comment (mosesn) [inline doc/src/sphinx/HttpStreaming.rst]: we need to close the response here to see an EOF

**Comment by mosesn** [comment]
@luciferous are you still interested in working on this?  if not, I might ransack it for a quick explanation on how to use the Reader + Writer apis.

**Comment by luciferous** [comment]
I'll have a look this week.

**Comment by luciferous** [comment]
@jbripley There's an example HTTP streaming [client](https://github.com/twitter/finagle/blob/develop/finagle-example/src/main/scala/com/twitter/finagle/example/http/HttpStreamingClient.scala) and [server](https://github.com/twitter/finagle/blob/develop/finagle-example/src/main/scala/com/twitter/finagle/example/http/HttpStreamingServer.scala) in finagle/finagle-example.

(Sorry for such a delayed reply.)

**Comment by luciferous** [comment]
@mosesn I didn't have enough time to work on this last week. Don't let me delay you, if you're wanting to ransack for Reader + Writer stuff, do it. Also, there's an old gist about Streaming that was a WIP, please take a look: https://gist.github.com/luciferous/76dc78e225ebb80ae3db.

**Comment by mosesn** [comment]
@luciferous I'm not on guru duty anymore, but I might check in with this week's guru to see if they're interested in doing it.  Thanks!

**Comment by kristofa** [comment]
We enabled streaming for one of our services lately and got our use case working but it took us quite some effort to get it right. Adding more documentation will be useful and appreciated! I'll be happy to help reviewing.

**Comment by argha-c** [comment]
Indeed, as @kristofa mentions, it'd be nice to improve the documentation here. Happy to contribute in any way around this.

**Comment by bryce-anderson** [comment]
There hasn't been much activity on this and many parts are no longer valid. Barring any resurgent interest in getting this updated and merged, I'm going to close this at the end of the week.

**Comment by bryce-anderson** [comment]
Closing due to inactivity.

**Pull request closed without merging**